feat(eslint-plugin-rules): add eslint-plugin-rules (#465)
theforeman/foreman-js: packages/eslint-plugin-rules/README.md

# Eslint Plugin Rules

## About

Place for eslint custom rules for Foreman and plugins

## Usage

### Create an eslintrc file
```js
{
  "plugins": ["@theforeman/rules"],
  "rules": {
    "@theforeman/rules/require-ouiaid": "warn",
  }
}
```

> These components would be checked for ouiaId field if none is specified in the eslintrc file:

Reviewer: ```suggestion
If not specified in the eslintrc file, these components will be checked for an `ouiaId` prop:
```

```
  "Alert",
  "Breadcrumb",
  "Button",
  "Card",
  "Checkbox",
  "Chip",
  "ChipGroup",
  "ContextSelector",
  "Dropdown",
  "DropdownItem",
  "DropdownSeparator",
  "DropdownToggle",
  "DropdownToggleCheckbox",
  "FormSelect",
  "Menu",
  "Modal",
  "ModalContent",
  "Nav",
  "NavExpandable",
  "NavItem",
  "OptionsMenu",
  "Pagination",
  "Radio",
  "Select",
  "Switch",
  "TabButton",
  "TabContent",
  "Tabs",
  "Text",
  "TextInput",
  "Title",
  "Toolbar",
  "Table",
  "TableComposable",
  "Tr"
```

You can specify what components you want to check against.
```js
{
  "plugins": ["@theforeman/rules"],
  "rules": {
    "@theforeman/rules/require-ouiaid": [
      "warn",
      "Button",

>       "Table",

Reviewer: Is the warning level really the first element in the array with all the other components? That seems odd..

Author: Yep, that is the [format](https://eslint.org/docs/latest/extend/custom-rules#accessing-options-passed-to-a-rule).

# Eslint Plugin Rules

## About

Place for eslint custom rules for Foreman and plugins

## Usage

### Create an eslintrc file
```js
{
  "plugins": ["@theforeman/rules"],
  "rules": {
    "@theforeman/rules/require-ouiaid": "warn",
  }
}
```

> If not specified in the eslintrc file, these components will be checked for an `ouiaId` prop:

Reviewer 2: Can you add here why we need an `ouiaId` prop in these components, and an example?

Author: `ouiaId` field is used for QE's automation tests.

Author: @MariaAga Updated.

Reviewer 2: Thanks
